FreeRTOS/FreeRTOS-Kernel
# Wrap macros in do { ... } while( 0 )

Wrap macros in do { ... } while( 0 )

Description
-----------
If statements in macros can be hazardous. Here's a contrived example:

    if( length == 0 )
        configASSERT( data == NULL )
    else
        configASSERT( data != NULL )

This looks sensible at first glance, but it expands to:

    if( length == 0 )
        if (!( data == NULL )) { ... }
    else
        if (!(data != NULL )) { ... }

... which is not intended; the `else` block associates with the `if` from the assert macro instead of the `if( length == 0)`. The usual pattern to avoid this is to wrap the macro in `do { ... } while (0)` which causes the right thing to happen.

Test Steps
-----------
n/a

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
